.github/workflows/docker-publish.yml

```
@@ -0,0 +1,104 @@
+# This is a basic workflow to help you get started with Actions
+
+name: Docker Build and Publish
+
+# Controls when the action will run. Triggers the workflow on push or pull request
+# events but only for the main branch
+on:
+  schedule:
+    - cron: '0 0 * * *' # Nightly build
+  pull_request:
+    branches: [ master ] 
+  push:
+    branches: [ master ]
+
+  workflow_dispatch: 
+    
+# A workflow run is made up of one or more jobs that can run sequentially or in parallel
+jobs:
+  # This workflow contains a single job called "build"
+  build:
```

```suggestion
  build_and_push:
    name: Build, tag, and push images to sopelirc/sopel
    if: ${{ github.repository_owner == 'sopel-irc' }}
```

1. Better (but not necessarily complete) description of whats actually happening in this job
2. Only run this job for this repo… Forks will have to append a commit and adjust their `.gitconfigs` to something like:
```gitconfig
[pull]
	rebase = merges
[rebase]
	autostash = true
```

---

What good logic is there banning forks from using GHA?

---

My goal was not to ban forks, but instead to prevent a whole slew of failed actions from piling up in forks that didn't need/plan/have the set-up to deploy built images. For example, if I naively forked this repo, I would start seeing failed actions for this workflow each night, since I didn't prepare the necessary secrets.

So maybe a better solution is:, the `if` should check that all 3 required secrets are not empty. If any one of the secrets is empty, there is no need to run the workflow since the  step would fail anyway.

```suggestion
  build_and_push:
    name: Build, tag, and push images to sopelirc/sopel
    if: ${{ (secrets.DOCKER_SOPEL_IMAGE_NAME != '') && (secrets.DOCKER_HUB_USERNAME != '') && (secrets.DOCKER_HUB_ACCESS_TOKEN != '') }}
```